### PR TITLE
PPC build fixes

### PR DESCRIPTION
**Pull request by martin-schwenke**
2 of these commits should fix the (big-endian and 32-bit PPC) build problems seen in Issue #15923.  The other seems to be required to build with an ancient version of binutils.

Testing:
* Tested on IBM 64-bit little-endian Power 9 LPAR with gcc-11.1.1 and binutils-2.35.1. Builds, runs (`LD_LIBRARY_PATH=. openssl speed ecdhp384 ecdsap384`), no visible performance penalty due to the change of branch instruction.
* Tested on IBM 64-bit big-endian Power 7 LPAR with gcc-8.3.1 and binutils-2.29.1. Builds, runs .
* The 32-bit (powerpc) fix is untested. However, it seems quite obvious given that building the code is conditional on 64-bit but the function calls were built unconditionally.
* I'm unable to test the build on AIX because the compiler is complaining about type conflicts between gcc and system headers (even on code that built fine 2 weeks ago). Not sure what has changed on this machine... sorry about that!

I'm running `make test` on the test machines I have and will confirm results in a comment a bit later...

### CODE REVIEW

**Comment by martin-schwenke** [comment]
On P9 little-endian:
```
...
All tests successful.
Files=240, Tests=3245, 500 wallclock secs ( 6.82 usr  0.42 sys + 465.25 cusr 34.08 csys = 506.57 CPU)
Result: PASS
```

File: crypto/bn/bn_ppc.c [hunk 1]
```diff
@@ -40,12 +40,14 @@ int bn_mul_mont(BN_ULONG *rp, const BN_ULONG *ap, const BN_ULONG *bp,
      * no opportunity to figure it out...
      */
 
+#if _ARCH_PPC64
```
Comment (t8m) [inline crypto/bn/bn_ppc.c:43]: Is this universally available? Should it be `#ifdef _ARCH_PPC64` instead?

**Comment by martin-schwenke** [comment]
On P7 big-endian:
```
...
All tests successful.
Files=240, Tests=3245, 694 wallclock secs ( 8.50 usr  0.67 sys + 627.21 cusr 59.46 csys = 695.84 CPU)
Result: PASS
```

File: crypto/bn/bn_ppc.c (hunk 1 above)
Comment (martin-schwenke) [reply crypto/bn/bn_ppc.c:43]: Good catch.  Thanks...

Other places use `#if defined(_ARCH_PPC64)` so I'll use that for consistency.

**Comment by martin-schwenke** [comment]
@t8m Do you have any idea what's happening in https://github.com/openssl/openssl/issues/15923#issuecomment-872202503 ?

The only differences I see between my branch and `master` are:
```
$ git fetch origin
$ git diff --stat origin/master
 crypto/bn/asm/ppc64-mont-fixed.pl   | 4 ++--
 crypto/bn/bn_ppc.c                  | 2 ++
 doc/man7/openssl_user_macros.pod.in | 6 ++----
 3 files changed, 6 insertions(+), 6 deletions(-)
```
I'm confused how these changes could cause test failures on AIX.

Thanks for any clues...

**Comment by t8m** [comment]
> @t8m Do you have any idea what's happening in [#15923 (comment)](https://github.com/openssl/openssl/issues/15923#issuecomment-872202503) ?

He most probably built master branch from your repo and not the master-fixes branch.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for these fixes.

**Pull request closed without merging**